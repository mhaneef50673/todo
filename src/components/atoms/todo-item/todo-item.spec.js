import React from 'react';
import { mount } from 'enzyme';
import { mountToJson} from 'enzyme-to-json';
import TodoItem from './index';

describe('TodoItem Component', () => {
  const onTodoItemClickHandler = jest.fn();
  it('should render the component', () => {
    let tree = mount(
      <TodoItem
        name="Item 1"
        status="PENDING"
        index={0}
        onTodoItemClickHandler={onTodoItemClickHandler} 
      />
    )
    expect(mountToJson(tree)).toMatchSnapshot();
    tree = mount(
      <TodoItem
        name="Item 1"
        status="COMPLETED"
        index={0}
        onTodoItemClickHandler={onTodoItemClickHandler} 
      />
    )
    expect(mountToJson(tree)).toMatchSnapshot();
  });
  it('should run onTodoItemClickHandler on click', () => {
    const tree = mount(
      <TodoItem
        name="Item 1"
        status="PENDING"
        index={0}
        onTodoItemClickHandler={onTodoItemClickHandler} 
      />
    )
    const div = tree.find('.todo-item');
    div.simulate('click');
    expect(onTodoItemClickHandler).toBeCalled();
  })
});