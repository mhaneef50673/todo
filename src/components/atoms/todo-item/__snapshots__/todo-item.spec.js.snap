// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem Component should render the component 1`] = `
<Memo(TodoItem)
  index={0}
  name="Item 1"
  onTodoItemClickHandler={[MockFunction]}
  status="PENDING"
>
  <div
    className="todo-item PENDING"
    onClick={[Function]}
  >
    <span
      className="text"
    >
      Item 1
    </span>
    <span
      className="text"
    >
      PENDING
    </span>
  </div>
</Memo(TodoItem)>
`;

exports[`TodoItem Component should render the component 2`] = `
<Memo(TodoItem)
  index={0}
  name="Item 1"
  onTodoItemClickHandler={[MockFunction]}
  status="COMPLETED"
>
  <div
    className="todo-item COMPLETED"
    onClick={[Function]}
  >
    <span
      className="text"
    >
      Item 1
    </span>
    <span
      className="text"
    >
      COMPLETED
    </span>
  </div>
</Memo(TodoItem)>
`;
